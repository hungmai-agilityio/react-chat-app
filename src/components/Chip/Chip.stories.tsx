import { Meta, StoryObj } from '@storybook/react';

// FontAwesome
import { faCheckCircle, faTrash } from '@fortawesome/free-solid-svg-icons';

// Components
import Chip from '@/components/Chip';

const meta: Meta<typeof Chip> = {
  component: Chip,
  tags: ['autodocs'],
  argTypes: {
    children: {
      description: 'Render dynamic react node children'
    },
    styles: {
      description: 'Add custom css'
    },
    icon: {
      description: 'Render icon with FontAwesome'
    },
    onClick: {
      description: 'Handle event when click '
    }
  }
};

export default meta;

type Story = StoryObj<typeof Chip>;

export const Default: Story = {
  args: {
    children: 'June, 04 2024'
  }
};

export const ChipWithIcon: Story = {
  args: {
    children: 'Checked',
    icon: faCheckCircle
  }
};

export const ChipCustom: Story = {
  args: {
    children: 'Delete',
    icon: faTrash,
    styles: 'text-white bg-red-400',
    onClick: () => {
      alert('Remove?');
    }
  }
};
