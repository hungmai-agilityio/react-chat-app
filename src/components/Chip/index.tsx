import { ReactNode } from 'react';
import { clsx } from 'clsx';

// FontAwesome
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

interface ChipProps {
  children: ReactNode;
  styles?: string;
  icon?: IconDefinition;
  onClick?: () => void;
}

const Chip = ({ children, styles, icon, onClick }: ChipProps) => {
  return (
    <div
      onClick={onClick}
      className={clsx(
        'max-w-40 text-center truncate py-2 px-4 rounded-3xl shadow-chip font-medium text-md bg-white text-dark cursor-pointer',
        styles
      )}
    >
      {children}
      {icon && <FontAwesomeIcon icon={icon} className="ml-5" />}
    </div>
  );
};

export default Chip;
