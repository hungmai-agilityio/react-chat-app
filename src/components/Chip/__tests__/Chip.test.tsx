import { fireEvent, render, screen } from '@testing-library/react';

// Components
import Chip from '@/components/Chip';
import { faCheck, faTrash } from '@fortawesome/free-solid-svg-icons';

describe('Chip', () => {
  it('render Chip component with default props', () => {
    render(<Chip>June, 04 2024</Chip>);

    const element = screen.getByText('June, 04 2024');
    expect(element).toBeInTheDocument();
  });

  it('render Chip component with icon', () => {
    const { container } = render(<Chip icon={faCheck}>Check</Chip>);

    const svgElement = container.querySelector('.fa-check');
    expect(svgElement).toBeInTheDocument();
  });

  it('render Chip component with custom', () => {
    const props = {
      children: 'Delete',
      icon: faTrash,
      styles: 'text-white bg-red-400',
      onClick: jest.fn()
    };
    const { container } = render(<Chip {...props} />);

    const svgElement = container.querySelector('.fa-trash');
    expect(svgElement).toBeInTheDocument();

    const element = screen.getByText('Delete');

    fireEvent.click(element);
    expect(props.onClick).toHaveBeenCalled();
  });

  it('should render match to snapshot', () => {
    const { container } = render(<Chip>June, 04 2024</Chip>);
    expect(container).toMatchSnapshot();
  });
});
