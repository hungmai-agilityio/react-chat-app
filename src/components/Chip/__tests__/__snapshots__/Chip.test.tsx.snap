// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip should render match to snapshot 1`] = `
<div>
  <div
    class="max-w-40 text-center truncate py-2 px-4 rounded-3xl shadow-chip font-medium text-md bg-white text-dark cursor-pointer"
  >
    June, 04 2024
  </div>
</div>
`;
